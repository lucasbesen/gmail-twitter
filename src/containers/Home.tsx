import React from 'react';
import { useQuery } from '@apollo/react-hooks';
import gql from 'graphql-tag';

interface User {
  name: string;
}

interface Tweet {
  created_at: Date;
  id: string;
  text: string;
  user: User;
}

interface Twitter {
  search: Tweet[];
}

interface TweetsQueryData {
  twitter: Twitter;
}

interface TweetsQueryVars {
  q: string;
  count: number;
  resultType: string;
}

const TWEETS_QUERY = gql`
  query TweetsQuery($q: String!, $count: Int!, $resultType: SearchReponse!) {
    twitter {
      search(q: $q, count: $count, result_type: $resultType) {
        id
        text
        created_at
        user {
          name
        }
      }
    }
  }
`;

const Home: React.FC = () => {
  const { data, loading } = useQuery<TweetsQueryData, TweetsQueryVars>(TWEETS_QUERY, {
    variables: { q: '#mars', count: 40, resultType: 'recent' },
  });

  return <div>testing</div>;
};

export default Home;
