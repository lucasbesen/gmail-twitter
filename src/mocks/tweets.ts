import { ITweet } from '../types';

export const tweetMock: ITweet = {
  id: '11111111',
  text: 'This is a tweet',
  user: {
    name: 'User number 1',
  },
  created_at: 'Wed Aug 21 23:17:05 +0000 2019',
};
