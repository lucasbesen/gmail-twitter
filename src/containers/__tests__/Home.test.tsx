import React from 'react';
import renderer, { act } from 'react-test-renderer';
import { MockedProvider } from '@apollo/react-testing';
import { MemoryRouter } from 'react-router';
import wait from 'waait';

import Home, { TWEETS_QUERY } from '../Home';
import { tweetMock, tweetMock2 } from '../../mocks/tweets';

const mocks = [
  {
    request: {
      query: TWEETS_QUERY,
      variables: {
        q: '#mars',
        count: 40,
        resultType: 'recent',
      },
    },
    result: {
      data: {
        twitter: {
          search: [...[tweetMock], ...[tweetMock2]],
        },
      },
      loading: false,
    },
  },
];

it('should render Home with loading', async () => {
  const component = renderer.create(
    <MockedProvider mocks={mocks} addTypename={false}>
      <MemoryRouter>
        <Home />
      </MemoryRouter>
    </MockedProvider>,
  );
  const tree = component.toJSON();

  expect(tree).toMatchSnapshot();
});

it('should render Home without error', async () => {
  await act(async () => {
    const component = renderer.create(
      <MockedProvider mocks={mocks} addTypename={false}>
        <MemoryRouter>
          <Home />
        </MemoryRouter>
      </MockedProvider>,
    );
    await wait(0);
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });
});
