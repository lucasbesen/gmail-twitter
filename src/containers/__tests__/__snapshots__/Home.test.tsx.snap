// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Home with loading 1`] = `
<div
  className="sc-jTzLTM dcDWFI"
>
  <div
    className="sc-fjdhpX ctwrhO"
  >
    <svg
      aria-hidden="true"
      className="sc-bwzfXH ldVAyv StyledIconBase-sc-bdy9j4 jbPeIR"
      fill="currentColor"
      focusable="false"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M7 5c-1.103 0-2 .897-2 2v10c0 1.103.897 2 2 2h10c1.103 0 2-.897 2-2V7c0-1.103-.897-2-2-2H7zm0 12V7h10l.002 10H7z"
      />
    </svg>
    <svg
      aria-hidden="true"
      className="sc-ifAKCX exhDBy StyledIconBase-sc-bdy9j4 jbPeIR"
      fill="currentColor"
      focusable="false"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M17.65 6.35A7.958 7.958 0 0012 4c-4.42 0-7.99 3.58-7.99 8s3.57 8 7.99 8c3.73 0 6.84-2.55 7.73-6h-2.08A5.99 5.99 0 0112 18c-3.31 0-6-2.69-6-6s2.69-6 6-6c1.66 0 3.14.69 4.22 1.78L13 11h7V4l-2.35 2.35z"
      />
    </svg>
    <svg
      aria-hidden="true"
      className="sc-EHOje eRdHEU StyledIconBase-sc-bdy9j4 jbPeIR"
      fill="currentColor"
      focusable="false"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M12 8c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zm0 2c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm0 6c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z"
      />
    </svg>
  </div>
  <div
    className="sc-VigVT cCjTUW"
  >
    <div
      className="MuiCircularProgress-root MuiCircularProgress-colorSecondary MuiCircularProgress-indeterminate"
      role="progressbar"
      style={
        Object {
          "height": 40,
          "width": 40,
        }
      }
    >
      <svg
        className="MuiCircularProgress-svg"
        viewBox="22 22 44 44"
      >
        <circle
          className="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
          cx={44}
          cy={44}
          fill="none"
          r={20.2}
          strokeWidth={3.6}
          style={Object {}}
        />
      </svg>
    </div>
  </div>
</div>
`;

exports[`should render Home without error 1`] = `
<div
  className="sc-jTzLTM dcDWFI"
>
  <div
    className="sc-fjdhpX ctwrhO"
  >
    <svg
      aria-hidden="true"
      className="sc-bwzfXH ldVAyv StyledIconBase-sc-bdy9j4 jbPeIR"
      fill="currentColor"
      focusable="false"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M7 5c-1.103 0-2 .897-2 2v10c0 1.103.897 2 2 2h10c1.103 0 2-.897 2-2V7c0-1.103-.897-2-2-2H7zm0 12V7h10l.002 10H7z"
      />
    </svg>
    <svg
      aria-hidden="true"
      className="sc-ifAKCX exhDBy StyledIconBase-sc-bdy9j4 jbPeIR"
      fill="currentColor"
      focusable="false"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M17.65 6.35A7.958 7.958 0 0012 4c-4.42 0-7.99 3.58-7.99 8s3.57 8 7.99 8c3.73 0 6.84-2.55 7.73-6h-2.08A5.99 5.99 0 0112 18c-3.31 0-6-2.69-6-6s2.69-6 6-6c1.66 0 3.14.69 4.22 1.78L13 11h7V4l-2.35 2.35z"
      />
    </svg>
    <svg
      aria-hidden="true"
      className="sc-EHOje eRdHEU StyledIconBase-sc-bdy9j4 jbPeIR"
      fill="currentColor"
      focusable="false"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M12 8c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zm0 2c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm0 6c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z"
      />
    </svg>
  </div>
  <div
    className="sc-iwsKbI jsRrR"
  >
    <svg
      aria-hidden="true"
      className="sc-bwzfXH ldVAyv StyledIconBase-sc-bdy9j4 jbPeIR"
      fill="currentColor"
      focusable="false"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M7 5c-1.103 0-2 .897-2 2v10c0 1.103.897 2 2 2h10c1.103 0 2-.897 2-2V7c0-1.103-.897-2-2-2H7zm0 12V7h10l.002 10H7z"
      />
    </svg>
    <svg
      aria-hidden="true"
      className="sc-bZQynM htugFp StyledIconBase-sc-bdy9j4 jbPeIR"
      fill="currentColor"
      focusable="false"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M22 9.24l-7.19-.62L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21 12 17.27 18.18 21l-1.63-7.03L22 9.24zM12 15.4l-3.76 2.27 1-4.28-3.32-2.88 4.38-.38L12 6.1l1.71 4.04 4.38.38-3.32 2.88 1 4.28L12 15.4z"
      />
    </svg>
    <div
      className="sc-gZMcBi NNoYB"
    >
      User number 1
    </div>
    <div
      className="sc-gqjmRU dFAKmp"
    >
      This is a tweet
    </div>
    <div
      className="sc-gZMcBi NNoYB"
    >
      21/08/2019 20:17
    </div>
  </div>
  <div
    className="sc-iwsKbI jsRrR"
  >
    <svg
      aria-hidden="true"
      className="sc-bwzfXH ldVAyv StyledIconBase-sc-bdy9j4 jbPeIR"
      fill="currentColor"
      focusable="false"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M7 5c-1.103 0-2 .897-2 2v10c0 1.103.897 2 2 2h10c1.103 0 2-.897 2-2V7c0-1.103-.897-2-2-2H7zm0 12V7h10l.002 10H7z"
      />
    </svg>
    <svg
      aria-hidden="true"
      className="sc-bZQynM htugFp StyledIconBase-sc-bdy9j4 jbPeIR"
      fill="currentColor"
      focusable="false"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M22 9.24l-7.19-.62L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21 12 17.27 18.18 21l-1.63-7.03L22 9.24zM12 15.4l-3.76 2.27 1-4.28-3.32-2.88 4.38-.38L12 6.1l1.71 4.04 4.38.38-3.32 2.88 1 4.28L12 15.4z"
      />
    </svg>
    <div
      className="sc-gZMcBi NNoYB"
    >
      User number 2
    </div>
    <div
      className="sc-gqjmRU dFAKmp"
    >
      This is a tweet 2
    </div>
    <div
      className="sc-gZMcBi NNoYB"
    >
      19/08/2019 20:17
    </div>
  </div>
</div>
`;
